src/EventStore.Core/Services/SubscriptionsService.cs

```diff
@@ -335,6 +350,69 @@ public class SubscriptionsService<TStreamId> :
 			return resolvedEvent;
 		}
 
+		private void ProcessStreamMetadataChanges(string eventStreamId) {
+			if (eventStreamId == AllStreamsSubscriptionId)
```

Comment:
It might be worth moving these conditions to ProcessEventCommitted so that this method isn't entered for every event

Reply:
i've moved `ProcessStreamMetadataChanges()` to `Handle(StorageMessage.EventCommitted message)` / `Handle(StorageMessage.InMemoryEventCommitted message)`. this gets rid of the unnecessary `AllStreamsSubscriptionId` empty string check

for the other checks (`SystemStreams.IsMetastream()` / `_subscriptionTopics.TryGetValue()`), i've kept them in the ProcessStreamMetadataChanges(). even though it's on a hot path, i don't think that it should cause any performance penalty as entering/leaving the method is negligible and the method will also be inlined automatically by the JIT compiler if it's called frequently.

Reply:
it also raises the question as to whether or not we should process stream metadata changes for ephemeral streams. i've tested adding stream metadata to the $mem-node-state ephemeral stream and it's working fine.

Reply:
the memstream thing is fine by me

(moving the conditions was my idea. i was thinking the target method might be on the large side for inlining and and the checks will almost always return false it seemed worth it,   but now that we're calling it from two places, and you're right its probably not going to make much practical diference, lets leave it as is)